tensorflow/contrib/android/java/org/tensorflow/contrib/android/TensorFlowInferenceInterface.java

```diff
@@ -50,6 +56,117 @@
   private static final String TAG = "TensorFlowInferenceInterface";
   private static final String ASSET_FILE_PREFIX = "file:///android_asset/";
 
+  /**
+   * Copies resource to a cache file.
+   * @param  context context The context that contains the files
+   * @param  String filename  Filename--take note that it checks for .pb in asset case
+   * @param  boolean res Is it from res/raw
+   *
+   * @return File reference to cache file.
+   */
+  private File createCacheFile(Context context, String filename, boolean res)
+      throws IOException {
+    InputStream inputStream = null;
+    if (res) {
+      int resourceId = context.getResources().getIdentifier(filename, "raw", context.getPackageName());
+      if (resourceId == 0) {
+        return null;
+      }
+      inputStream = context.getResources().openRawResource(resourceId);
+    } else {
+      if (!filename.endsWith(".pb")) {
+        filename += ".pb";
+      }
+      inputStream = context.getAssets().open(filename);
+    }
+    File cacheFile = new File(context.getCacheDir(), filename);
+    if (cacheFile.createNewFile() == false) {
+      cacheFile.delete();
+      cacheFile.createNewFile();
+    }
+    FileOutputStream fileOutputStream = new FileOutputStream(cacheFile);
+    byte[] buffer = new byte[1024 * 512];
+    int count;
+    while ((count = inputStream.read(buffer)) != -1) {
+      fileOutputStream.write(buffer, 0, count);
+    }
+    fileOutputStream.close();
+    inputStream.close();
+    return cacheFile;
+  }
+
+  /*
+  * Load a TensorFlow model via FileChannel mmap
+  * This requires that assets files are copied to internal storage already
+  * getFileStreamPath() points to that directory; context used must be passed in here
+  *
+  * @param ctx The context to use to load the model file; same as one used to copy files
+  * @param model The filepath to the GraphDef proto representing the model.
+  */
+  public TensorFlowInferenceInterface(Context ctx, String modelName) {
+    prepareNativeRuntime();
+    File modelFile = null;
+    this.g = new Graph();
+    this.sess = new Session(g);
+    this.runner = sess.runner();
+    this.modelName = modelName;
+    MappedByteBuffer buffer = null;
+    FileChannel fileChannel = null;
+    RandomAccessFile raf = null;
+    File cacheFile = null;
+    try {
+      cacheFile = createCacheFile(ctx, modelName, true);
+      if (cacheFile != null) {
+        //maybe do fis
```

Comment:
Space after // and please spell out the acronym

Reply:
@andrewharp I will remove this